File: Safari-611.1.10/LayoutTests/http/tests/misc/resources/basic-echo-post.php
<?php
header("Cache-Control: no-store");
header("Connection: close");
if (!isset($_SERVER['PHP_AUTH_USER']) 
    || !isset($_SERVER['PHP_AUTH_PW'])) {
    header("WWW-Authenticate: Basic realm=\"TestRealm\"");
    header('HTTP/1.0 401 Unauthorized');
    echo "Please send a username and password";
    exit;
}

$myfile = fopen($_FILES["data"]["tmp_name"], "r") or die("Unable to open file!");
echo "Uploaded blob data: ";
echo fread($myfile,$_FILES["data"]["size"]);
fclose($myfile);
?>

File: LayoutTests/http/tests/misc/resources/basic-echo-post.py
#!/usr/bin/env python3

import base64
import cgi
import os
import sys

credentials = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')
username = credentials[0]
password = ':'.join(credentials[1:])

sys.stdout.write(
    'Cache-Control: no-store\r\n'
    'Connection: close\r\n'
    'Content-Type: text/html\r\n'
)

if not username or not password:
    sys.stdout.write(
        'WWW-Authenticate: Basic realm=\"TestRealm\"\r\n'
        'status: 401\r\n\r\n'
        'Please send a username and password'
    )
else:
    form = cgi.FieldStorage()
    sys.stdout.write(
        '\r\n'
        'Uploaded blob data: {}'.format(str(form.getvalue('data'), 'utf-8'))
    )